text/0000-or-patterns.md

```diff
@@ -0,0 +1,666 @@
+- Feature Name: `or_patterns`
+- Start Date: 2018-08-29
+- RFC PR: _
+- Rust Issue: _
+
+# Summary
+[summary]: #summary
+
+Allow `|` to be arbitrarily nested within a pattern such
+that `Some(A(0) | B(1 | 2))` becomes a valid pattern.
+The operator `|` has low precedence such that `i @ p | q` means `(i @ p) | q`.
```

Comment:
Can you cross-reference the explanation of this precedence? My first instinct was to prefer to see `@` as the lowest-precedence thing in patterns, unless there's some specific compatibility reason to do otherwise, but at the same time I could readily be convinced otherwise if you have a rationale for this.

Also, I don't think you need to put the precedence in the summary; you can put it later in the document.